config/dryrun/dryrun_rbac.yaml

```diff
@@ -0,0 +1,27 @@
+# Additional permission is required to enable DryRun.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: manifest-verify-dry-run
+  namespace: kyverno
+rules:
+- apiGroups:
+  - '*'
+  resources:
+  - '*'
```

Comment:
Hi @rurikudo - for Kyverno, we follow a "least privileged" approach and only provide the minimal permissions required during the installation.

Administrators can then add more more permissions as needed. 

Hence, can we:

1. Remove this wildcard permission
2. Add a validation check such that when "dry-run" is enabled for a resource we check if that resource can be created by Kyverno, and return an error if it cannot be.

There is a similar check for the generate policy rule, as it has the same challenge.

See: https://github.com/kyverno/kyverno/blob/41a3f6c3889b3629b5c46ed9ce5cc4baff5eccef/pkg/policy/generate/auth.go#L37

Reply:
Hi @JimBugwadia , I have updated the role to follow least privileged approach. Also, I added the permission check for dry-run. Thank you.